src/Metric.jl

```diff
@@ -78,41 +101,48 @@ Determinant of local matrix representation of the metric tensor $g$
 """
 det_local_metric(M::MetricManifold, x) = det(local_metric(M, x))
 
-inner(M::MetricManifold, x, v, w) = dot(v, local_metric(M, x) * w)
+@traitfn inner(M::MMT, x, v, w) where {MT<:Manifold,GT<:Metric,MMT<:MetricManifold{MT,GT};!HasMetric{MT,GT}} = dot(v, local_metric(M, x) * w)
+@traitfn inner(M::MMT, x, v, w) where {MT<:Manifold,GT<:Metric,MMT<:MetricManifold{MT,GT};HasMetric{MT,GT}} = inner(M.manifold, x, v, w)
+
```

Comment:
That's a nice idea, because it enables the user to specify a performant default metric (with `MT`) that Is then just transparently passed through 👍

Reply:
Yeah this satisfies the two ideals we were discussing earlier. It keeps it simple to implement a new manifold without even thinking explicitly about metrics (but obviously assuming some sensible default), and if one then wants to implement several metrics, one can use `HasMetric` to designate the default that was assumed for the original manifold.

For now `local_metric` and its relatives take `MetricManifold`s only, since if you're implementing that, you're definitely thinking about metrics and should be working with a `Metric` and `MetricManifold`. And that gets you numerically integrated geodesics and the other tensors/curvature.